### ISSUE TITLE
[Reasing] PLaMo-100B: A Ground-Up Language Model Designed for Japanese Proficiency

### ISSUE THREAD

**Issue by ShotaSato01** [ShotaSato01/Paper_survey#105]
## 0. Paper
http://arxiv.org/abs/2410.07563

## 1. What is it?
PLaMo-100B：日本語能力の向上を目的として設計されたLLM
→2兆トークンを用いてスクラッチから学習された (pre：1.5兆  post：0.5兆)

2024年のGENIACプロジェクトの一環として1000億パラメータを有するLLMのPLaMo-100Bを開発した
→日本語タスクで最適な性能を発揮するように設計されたおり，同時に英語においても高い熟達度を実現する，スクラッチモデル

### Pre-Training

**Dataset**

<img width="1456" height="260" alt="Image" src="https://github.com/user-attachments/assets/1f620a24-9389-4c7d-8da3-a5776a787458" />
  
1)  WARCファイルをダウンロードし，日本語のHTMLおよびテキストファイルを抽出する
2) HTML形式の場合，抽出データを用いてMarkdown形式に変換する
3) llm-jp corpus filterを用いて文書全体をフィルタリングする
4) MinHashアルゴリズムを用いて，全てのダンプかんで重複削除を実施する
5) データセットとして利用可能なように，均一のサイズのチャンクへ再シャーディング

**Model Training Stabilization**
モデル学習の安定化のために，モデルアーキテクチャと損失関数をいう二つの主要な戦略を採用した

・QK Normalization
→PLaMp-100BのアーキテクチャはLlama2, Llama3に非常に近い構成
→自己注意層内の計算を安定化し，モデル全体の学習安定性向上に寄与するとされている

・Z-loss
→Softmaxクロスエントロピー損失の数値的安定性を向上させるz-lossを導入

<img width="790" height="225" alt="Image" src="https://github.com/user-attachments/assets/343c7d0f-2d46-4db0-bba7-c1a90e994d28" />
  
→学習進行のモニタリング指標として有用であることがわかった

**Performance Optimization**
事前学習開始時点で，H100GPUによって，1000億パラメータ級のLLMを学習した確率事例は存在していなかった
→様々な学習速度向上アプローチをとった
・ZeRo Bubble
→パイプライン並列化にはZeroBubbleを採用することで，GPUが待機状態になる時間を最小化することを期待

・Numerical Precision of the lm-head
<img width="808" height="505" alt="Image" src="https://github.com/user-attachments/assets/63297975-a51e-430d-bffe-16a489f52c70" />
  
→FP8による学習では，Z-lossが顕著に大きくなる

### Post-Training
**Post-training pipeline**

<img width="1318" height="389" alt="Image" src="https://github.com/user-attachments/assets/57c737c8-0710-4fae-9795-64c7045309ad" />

・Supervised Fine-Tuning
→「LLMに対する質問」と「LLMが出力すべき期待回答」のペアからなる事前定義データセットを用いて学習を行う
→SFTの学習は次トークン予測で行われるが，応答生成部分の微調整に重点を置く

先行研究にならい，二段階SFTを採用
1) 数学問題
2) その他の多様なタスク

・Direct Preference Optimization
→同じ質問に対する二つの応答を「より良い」または「より悪い」とラベル付けしたデータから人間の選好を学習する
オンポリシー：学習対象モデル自身が応答を生成し，それをラベル付け
オフポリシー：学習対象とはモデルが生成した応答を使用

SPIN：STFデータセットの教示応答とモデル生成応答をペアにし，「教示応答の方が望ましい」と仮定してデータセットを構築するハイブリット手法

本研究でのDPO構成
1) 公開選好データセット
2) PLaMo-100Bのスナップショットから生成したラベル付けをしたデータ
3) SPIN手法で生成したデータ

・Model Merging
→複数モデルを統合して全体性能を向上させる技術
Spherical Linear Interpolation：二つのモデル間の中間点を球面線形補完で計算する

**Post-training Data**
・公開データ
・プログラムによるデータ生成
・合成データ生成

選好データの生成には，PLaMo-100Bから8種類の応答を生成し，それぞれにスコアをつけた，
→スコアの最も良いものをChosen，スコアの最も悪いものをrejectedにした

### Evaluation Results

**Jasterデータセット**
→日本語能力を測定するための日本語ベンチマーク群
・NLI：自然言語推論
・QA：質問応答
・RC：読解
・MR：数学的推論
・FA：基礎分析

<img width="1459" height="794" alt="Image" src="https://github.com/user-attachments/assets/c05eb08a-8662-4a9c-99cd-9b03d52d86a1" />
  
→平均スコアでGPT_4を上回った
　事後学習において，多様な質問形式への応答方法を学習することで，事前学習で獲得した知識を活用する能力が向上した

**MT-Bench**
→LLMの対話応答能力を評価するベンチマークで以下のカテゴリで評価
・codeing
・extraction
・humanities
・math
・reasoning
・roleplay
・STEM
・writing

<img width="1085" height="642" alt="Image" src="https://github.com/user-attachments/assets/c41b8dab-dc04-4f4a-a1ac-73471088c093" />

**Rakuda Benchmark**
→日本国内トピック(地理，政治，社会，歴史)に関する対話応答性能を評価するベンチマーク
・MT-Benchのjudgeプロンプトを用いた絶対評価
・公式ガイドラインプロンプロによる相対評価

<img width="1622" height="762" alt="Image" src="https://github.com/user-attachments/assets/abce0b79-306f-469d-a933-81546e2b8dc0" />

## 5. Is there a discussion?
GENIACプロジェクト期間中，社内の他チームではPLaMo-100Bの高い日本語能力を活かした，マルチモーダル拡張にも取り組んだ
→画像アダプタをを学習させることで，物体検出，画像分類，キャプション生成，VQAなど
